libdrgn/debug_info.c

```diff
@@ -539,10 +539,33 @@ LIBDRGN_PUBLIC bool drgn_module_address_range(const struct drgn_module *module,
 	return true;
 }
 
+static struct drgn_error *
+drgn_check_module_address_range_overlap(struct drgn_module *module,
+					uint64_t start, uint64_t end)
+{
+	for (auto it = drgn_module_address_tree_search_le(&module->prog->dbinfo.modules_by_address,
+							  &start);
+	     it.entry && it.entry->start < end;
+	     it = drgn_module_address_tree_next(it)) {
+		if (start < it.entry->end && it.entry->module != module) {
```

Comment:
Nice check to avoid raising an error if the new address ranges overlap with the old ones for the same module.